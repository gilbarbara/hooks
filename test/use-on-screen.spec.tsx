import * as React from 'react';
import { render } from '@testing-library/react';

import useOnScreen from '../src/use-on-screen';

describe('useOnScreen', () => {
  const Component = () => {
    const ref = React.useRef(null);
    const visible = useOnScreen(ref);

    return <div ref={ref}>{visible ? 'On Screen' : 'Offscreen'}</div>;
  };

  const { container } = render(<Component />);

  it('should render the is visible', () => {
    expect(container.firstChild).toMatchSnapshot();
  });
});
