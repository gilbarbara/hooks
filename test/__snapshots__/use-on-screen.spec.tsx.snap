// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useOnScreen should render the is visible 1`] = `
<div>
  Offscreen
</div>
`;
